turbopack/crates/turbopack-ecmascript/src/analyzer/graph.rs

```diff
@@ -196,6 +196,14 @@ pub enum Effect {
         span: Span,
         in_try: bool,
     },
+    /// A reference to `new Worker(...)`.
+    Worker {
+        url: JsValue,
+        options: Option<JsValue>,
+        ast_path: Vec<AstParentKind>,
+        span: Span,
+        in_try: bool,
+    },
```

Comment:
This is too specific. in `graph` you don't know the environment and you don't have complete info about other variables. You can't determine if `Worker` is the WebWorker object.

This decision need to be made during "linking", where you know about the environment and know about other variables.

Here is a case where this is important:

```
var W;

function run() {
  new W(new URL(...)
}

W = Worker;

run()
```

So instead we can only have a `NewCall` effect, which would be similar to `Cell`, but with a `new` prefix: `

Reply:
Doesn't the same argument also apply to `new URL`? That is currently also a `Effect::URL`

Reply:
yes